client/src/document/writer-toolbar/spy.tsx

```diff
@@ -26,7 +28,9 @@ export default function DocumentSpy({ onMessage }) {
       },
       onmessage: (e) => {
         const data = JSON.parse(e.data);
-        onMessage(data);
+        if (documentURL === data.documentUri) {
+          onDocumentUpdate();
+        }
```

Comment:
You have my blessing to edit the watcher so it's called `.documentURL` or `.mdn_url` instead. I think it's in the `triggerTouch` function.